mangaki/mangaki/management/commands/compare.py

```diff
@@ -1,121 +1,75 @@
-import csv
-import os.path
-from collections import Counter
-
-import numpy as np
 from django.conf import settings
 from django.core.management.base import BaseCommand
-from sklearn.metrics import mean_squared_error
-from sklearn.model_selection import train_test_split
 
+from collections import defaultdict
+from sklearn.model_selection import ShuffleSplit
+import numpy as np
+import csv
+import os.path
+
+from mangaki.utils.data import Dataset
 from mangaki.utils.wals import MangakiWALS
 from mangaki.utils.als import MangakiALS
 from mangaki.utils.knn import MangakiKNN
 from mangaki.utils.svd import MangakiSVD
 from mangaki.utils.pca import MangakiPCA
 from mangaki.utils.zero import MangakiZero
-
 from mangaki.utils.values import rating_values
 
-import matplotlib.pyplot as plt
 
-TEST_SIZE = 0.2  # 20% is kept for the test set
+NB_SPLIT = 5  # Divide the dataset into 5 buckets
 
 
 class Experiment(object):
-    X = None
-    y = None
-    X_test = None
-    y_test = None
-    y_pred = None
-    results = {}
-    algos = None
-    def __init__(self, PIG_ID=None):
-        self.algos = [MangakiALS(20), MangakiWALS(20), MangakiSVD(20), MangakiPCA(20), MangakiKNN(20), MangakiZero()]
-        # self.results.setdefault('x_axis', []).append()
-        self.make_dataset(PIG_ID)
-        self.execute()
-
-    def clean_dataset(self):
-        self.X = []
-        self.y = []
-        self.X_test = []
-        self.y_test = []
-        self.y_pred = []
+    def __init__(self, dataset_name):
+        self.algos = [
+            lambda: MangakiALS(20),
+            # lambda: MangakiWALS(20),
+            lambda: MangakiSVD(20),
+            # lambda: MangakiPCA(20),
+            # lambda: MangakiKNN(20),
+            # lambda: MangakiZero()
+        ]
+        self.anonymized = None
+        self.load_dataset(dataset_name)
+        self.compare_models()
```

Comment:
I think it's better to *not* compare models in the constructor.

Let it as a responsibility to the class user.

Otherwise, it will be impossible to construct an instance without comparing models right away.